examples/batch-predict/src/main/java/com/idibon/BatchPredict.java

```diff
@@ -0,0 +1,216 @@
+package com.idibon;
+
+import java.io.*;
+import java.util.Map;
+import java.util.stream.*;
+import java.nio.file.Files;
+import java.nio.charset.Charset;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.json.JsonObject;
+
+import com.idibon.api.http.impl.JdkHttpInterface;
+import com.idibon.api.model.*;
+import com.idibon.api.IdibonAPI;
+import com.idibon.api.util.Either;
+
+import org.apache.commons.cli.*;
+
+/**
+ * Iterates through documents in a collection and predicts every document
+ * against a task in parallel.
+ *
+ * Allows for command-line control of the number of parallel connections
+ * to use, and limiting the number of documents to a subset of the
+ * entire collection.
+ */
+public class BatchPredict {
+
+    public static void main(String[] args) throws Exception {
+        CommandLine options = parseCommandLine(args);
+        if (options == null) return;
+
+        IdibonAPI client = createClient(options);
+
+        /* count the number of classification operations performed by this
+         * application, to build a performance report at the end. */
+        AtomicInteger predictions = new AtomicInteger(0);
+        long start = System.currentTimeMillis();
+
+        try (Stream<DocumentContent> source = getSourceData(client, options)) {
```

Comment:
but on the broader point -- if you've been stuck with Java 6, definitely read up on 7 and 8. they're major and much-needed upgrades.